scapy/layers/inet6.py

```diff
@@ -374,8 +374,10 @@ def default_payload_class(self,p):
             return Raw
         elif self.nh == 135 and len(p) > 3: # Mobile IPv6
             return _mip6_mhtype2cls.get(ord(p[2]), MIP6MH_Generic)
+        elif self.nh == 43 and ord(p[2]) == 4:  # Segment Routing header
+            return IPv6ExtHdrSegmentRouting
```

Comment:
nh == 43 is already taken care of with get_cls(ipv6nhcls.get(self.nh, "Raw"), "Raw"). I forgot to mention it =/